src/urllib3/util/request.py

```diff
@@ -98,7 +99,9 @@ def make_headers(
     return headers
 
 
-def set_file_position(body, pos):
+def set_file_position(
+    body: Union[BinaryIO, TextIO], pos: Optional[Union[int, object]]
```

Comment:
Is there any advantage to using `Union[BinaryIO, TextIO]` over `IO`?

Reply:
Just to make it clear. I've replaced it with `IO[AnyStr]`